### PR TITLE
Changes for early fetching of methods

### PR DESCRIPTION
**Pull request by ricea**
Stream classes are changing to get methods from the underlying object
during construction rather than when the methods are called.

Change expectations of when method lookup happens.

See https://github.com/whatwg/streams/issues/691 for background.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21290)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/309549978)

File: streams/readable-streams/bad-underlying-sources.js [hunk 1]
```diff
@@ -82,9 +80,11 @@ promise_test(t => {
...
-      assert_object_equals(r, { value: 'a', done: false }, 'the chunk read should be correct');
+      assert_object_equals(r, { value: 'a', done: false }, 'the first chunk read should be correct');
     }),
-    promise_rejects(t, theError, reader.closed)
+    reader.read().then(r => {
+      assert_object_equals(r, { value: 'a', done: false }, 'the second chunk read should be correct');
+    })
   ]);
 
 }, 'Underlying source pull: throwing getter (second pull)');
```
Comment (domenic) [inline streams/readable-streams/bad-underlying-sources.js]: This test might benefit from a comment saying that the test is that it's only gotten once. Indeed, asserting that counter === 1 after the last assert_object_equals might be good.

File: streams/writable-streams/properties.js [hunk 2]
```diff
@@ -194,28 +194,32 @@ for (const method in sinkMethods) {
...
+    }
+    const sink = new Proxy({}, handler);
+    const ws = new WritableStream(sink);
+    assert_array_equals(touchedProperties, ['type', 'write', 'close', 'abort', 'start'],
+                        'expected properties should be got');
+    return trigger(ws.getWriter()).then(() => {
+      assert_array_equals(touchedProperties, ['type', 'write', 'close', 'abort', 'start'],
+                          'no properties should be accessed on method call');
+    });
+  }, `unexpected properties should not be accessed when calling sink method ${method}`);
```
Comment (domenic) [inline streams/writable-streams/properties.js]: Should we clone this test for readable too, while we're here?

(What a great test, BTW...)

**Review by domenic — APPROVED** [review]
Some suggestions, but overall looks good. I think we should add transform stream tests too though, probably before merging.

File: streams/writable-streams/properties.js (hunk 2 above)
Comment (ricea) [reply streams/writable-streams/properties.js]: I've been thinking about throwing away readable-stream's properties.js and use one based on this instead. But I'd have to check carefully that I wasn't losing any coverage, which is why I haven't done it yet.

File: streams/readable-streams/bad-underlying-sources.js (hunk 1 above)
Comment (ricea) [reply streams/readable-streams/bad-underlying-sources.js]: Done.

**Comment by ricea** [comment]
Corresponding standard change is https://github.com/whatwg/streams/pull/860 (still WIP at this time).

**Review by domenic — APPROVED** [review]
LGTM!

**Pull request merged**